src/Core/src/Platform/Gtk/ImageView.cs

```diff
@@ -9,11 +11,60 @@ namespace Microsoft.Maui.Platform
 
 	public class ImageView : Gtk.Image
 	{
+		// OriginalSizeBuf is added so when we want to resize the image using Pixbuf we can access the
+		// initial Pixbuf and resize that. 
+		Pixbuf? _originalSizeBuf;
+		// Image might be null because it is possible to have a WidthRequest before Pixbuf has a value.
+		// In this case, we'll just save the value, and once Pixbuf is initialized we'll update it's size
+		// to have the saved values
+		int _lastRequestedWidth = -1;
+		int _lastRequestedHeight = -1;
```

Comment:
@parhamsaremi what is this -1? if you want to represent uninitialized values, use `int?`

Reply:
hmmm let me try this.